Lib/test/test_ipaddress.py

```diff
@@ -96,10 +96,19 @@ def pickle_test(self, addr):
 class CommonTestMixin_v4(CommonTestMixin):
 
     def test_leading_zeros(self):
-        self.assertInstancesEqual("000.000.000.000", "0.0.0.0")
-        self.assertInstancesEqual("192.168.000.001", "192.168.0.1")
-        self.assertInstancesEqual("016.016.016.016", "16.16.16.16")
-        self.assertInstancesEqual("001.000.008.016", "1.0.8.16")
+        # bpo-36384: no leading zeros to avoid ambiguity with octal notation
+        msg = "Lead zeros are not permitted in '\d+'"
+        addresses = [
+            "000.000.000.000",
+            "192.168.000.001",
+            "016.016.016.016",
+            "192.168.000.001",
+            "001.000.008.016",
```

Comment:
It's unclear form these tests if leading zero is only disallowed in the first part of the IP (A in A.B.C.D) in all the 4 parts. I expect tests like:

* 0127.2.3.40
* 127.02.3.40
* 127.2.03.40
* 127.2.3.040

I chose these numbers since they are valid octal numbers: int("127", 8) = 87 ;-)

Reply:
``0127`` is already invalid because it's four characters. I added tests based on your suggestsion.